addons/keybinding/fnc_localizeKey.sqf

```diff
@@ -0,0 +1,54 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_localizeKey
+
+Description:
+    Translate DIK code and modifier keys to readable keybind.
+
+Parameters:
+    _key     - DIK code <NUMBER>
+    _shift   - Shift pressed? <BOOL>
+    _control - Control pressed? <BOOL>
+    _alt     - Alt pressed? <BOOL>
+
+Returns:
+    _keyName - Translated key <STRING>
+
+Examples:
+    (begin example)
+        [0x12, [false, false, false]] call CBA_fnc_localizeKey;
+    (end example)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+params [["_key", 0, [0]], ["_modifier", [], [[], false]]];
+
+if (_modifier isEqualType false) then {
+    _modifier = _this select [1, 3];
+};
+
+_modifier params [["_shift", false, [false]], ["_ctrl", false, [false]], ["_alt", false, [false]]];
+
+// Try to convert dik code to a human key code.
+private _keyName = GVAR(keyNames) getVariable str _key;
+
+if (isNil "_keyName") then {
+    _keyName = format [localize LSTRING(unkownKey), _key];
+};
+
+// Build the full key combination name.
+if (_alt && {!(_key in [DIK_LMENU, DIK_RMENU])}) then {
+    _keyName = localize "str_dik_alt" + "+" + _keyName;
```

Comment:
Totally minor, but is `[localize "bla", _keyName] joinString "+"` a faster method for these modifier keys?